### ISSUE TITLE
AccessTokenValidation with the middleware?

### ISSUE THREAD

**Issue by jonathh21** [IdentityServer/IdentityServer3#1786]
Hi,

I am extending an extisting webforms application that makes a call to wcf service with the username and password to authenticate, before then setting up lots of good stuff about the user.

I have integrated identity server v3 into this applications presentation layer and everything works well, however I want to pass the accesstoken to the wcf service rather than the username and password - so that the wcf layer can validate that token, and pull some claims down that would have previously been passed across with the name/password.

I am NOT planning on doing anything fancy to pass the access token across - one day I'll use a custom attribute or the likes but for now I just want to run some simple checks on the token, and the get resource claims as well as ideally set the identity principal too...

so basically I want to manually validate the accesstoken then call the userinfo endpoint to get the claims.

However the middleware is not an option for me as I can't use OWIN.

I am really struggling to find examples of this use case.. Any pointers please?

Thank you1!

**Comment by loctanvo** [comment]
Would this help you? https://gist.github.com/loctanvo/6c4cd684d2d4a2364eb0

Note the comments regarding access tokens.

**Comment by jonathh21** [comment]
Sorry that should read 'without the middleware'

Thanks for your comment but doesn't that relate to validating the identity token not the access token? Which endpoint is used for access token validation and how do you get the resource claims?

**Comment by leastprivilege** [comment]
Check the clients sample. The here's a wcf service and client. I also wrote a blog post recently.

**Comment by loctanvo** [comment]
@jonathh21 

>Thanks for your comment but doesn't that relate to validating the identity token not the access token? 

Maybe I misunderstood what you are trying to achieve. Probably the WCF sample @leastprivilege mentions can help. 

The code in the gist I posted can be used to validate access tokens manually, ie. validate the signature. This of course assumes the access token to be on JWT format.

**Comment by jonathh21** [comment]
Thanks guys - think that will help.. it is sods law.. I bust a gut getting what I have working over the weekend using the clients and samples from about 2 weeks ago.. .and 6 days ago the answer falls on my lap :)

@leastprivilege am I correct in that the code in your sample uses the jwttoken validator i.e. it DOESN'T ask the identity server to validate it? Also, how does one get the resource scope claims that ARE NOT in the access token?

**Comment by leastprivilege** [comment]
The sample assumes JWT - would be trivial to add reference token support to it.

For any other question - please open a new issue.
